### ISSUE TITLE
sustainable time for <Spin />

### ISSUE THREAD

**Issue by madcpt** [ant-design/ant-design#29568]
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
In many cases, we need a spinning icon to imply the refresh/reload action (which is what <Spin /> does). However, when refresh/reload time is too short to be noticed, we want to specify a minimum spinning time.

### What does the proposed API look like?
Specifies a minimum time in milliseconds for loading state (prevent blink)
```javascript
<Spin sustain={500}/>
```

**Comment by afc163** [comment]
Try `delay`, it is not exact same but has alternatively effect.

**Comment by madcpt** [comment]
> Try `delay`, it is not exact same but has alternatively effect.

I think `delay` specifies the minimun time ‘before’ spinning, while `sustain` indicates the minimun time ‘of’ spinning.

**Comment by zombieJ** [comment]
Agree. In my project, I usually wrap Spin & Button.loading with at lest 3 seconds to show loading to avoid fast flash. Maybe we can make delay as at least exist time instead?

**Comment by zombieJ** [comment]
Or
```tsx
<Spin delay={{ duration: 3000 }} />
```

**Comment by madcpt** [comment]
I would be happy to implement this feature :)

**Comment by madcpt** [comment]
> Or
> 
> ```tsx
> <Spin delay={{ duration: 3000 }} />
> ```

Do you think it's better to use another API like `sustain` instead of changing the old one? (it could raise error when users upgrade to future versions)

**Comment by zombieJ** [comment]
API naming should follow [naming standard](https://github.com/ant-design/ant-design/wiki/API-Naming-rules) and use the word which antd [already used](https://github.com/ant-design/ant-design/issues/16048).

Should find exist one instead of `sustain`

**Comment by jaylonez** [comment]
> Or
> 
> ```tsx
> <Spin delay={{ duration: 3000 }} />
> ```

This makes it sound like the delay duration is 3000 imo.

The naming standards document mentions minLength as an example prop which would be perfect I think but it hasn't been implemented in the 'current API standard' doc.

Of the already used prop names, I think 'duration' as it's own prop makes most sense, but it's locked, so I'm not sure if it's usable.

**Comment by madcpt** [comment]
I agree with @jaylonez, `duration` as its own prop also makes sense to me under current standard.

**Comment by zombieJ** [comment]
cc @ant-design/ant-design-collaborators

**Comment by 07akioni** [comment]
`minDuration`

**Comment by daribock** [comment]
I would love to see this feature. Because I am facing the same problem right now. Is there a work around to avoid fast flash?

**Comment by afc163** [comment]
@condorheroblog has a good point. https://github.com/ant-design/ant-design/issues/51828#issuecomment-2507036756
